deps/rabbit/src/rabbit_quorum_queue.erl

```diff
@@ -1524,3 +1529,25 @@ update_type_state(Q, Fun) when ?is_amqqueue(Q) ->
 overflow(undefined, Def) -> Def;
 overflow(<<"reject-publish">>, _Def) -> reject_publish;
 overflow(<<"drop-head">>, _Def) -> drop_head.
+
+-spec notify_decorators(amqqueue:amqqueue()) -> 'ok'.
+notify_decorators(Q) when ?is_amqqueue(Q) ->
+    QName = amqqueue:get_name(Q),
+    QPid = amqqueue:get_pid(Q),
+    case ra:local_query(QPid, fun rabbit_fifo:query_notify_decorators_info/1) of
+        {ok, {_, {MaxActivePriority, IsEmpty}}, _} ->
+            notify_decorators(QName, consumer_state_changed, [MaxActivePriority, IsEmpty]);
+        _ -> ok
+    end.
+
+notify_decorators(QName, Event) -> notify_decorators(QName, Event, []).
+
+notify_decorators(QName, F, A) ->
```

Comment:
This needs to run in a separate processes else it may crash or block the Ra leader.

Reply:
done